fastly/block_fastly_service_v1_logging_cloudfiles.go

```diff
@@ -162,33 +162,33 @@ func (h *CloudfilesServiceAttributeHandler) buildCreate(cloudfilesMap interface{
 
 	var vla = h.getVCLLoggingAttributes(df)
 	return &gofastly.CreateCloudfilesInput{
-		Service:           serviceID,
-		Version:           serviceVersion,
-		Name:              gofastly.NullString(df["name"].(string)),
-		BucketName:        gofastly.NullString(df["bucket_name"].(string)),
-		User:              gofastly.NullString(df["user"].(string)),
-		AccessKey:         gofastly.NullString(df["access_key"].(string)),
-		PublicKey:         gofastly.NullString(df["public_key"].(string)),
-		GzipLevel:         gofastly.Uint(uint(df["gzip_level"].(int))),
-		MessageType:       gofastly.NullString(df["message_type"].(string)),
-		Path:              gofastly.NullString(df["path"].(string)),
-		Region:            gofastly.NullString(df["region"].(string)),
-		Period:            gofastly.Uint(uint(df["period"].(int))),
-		TimestampFormat:   gofastly.NullString(df["timestamp_format"].(string)),
-		Format:            gofastly.NullString(vla.format),
-		FormatVersion:     vla.formatVersion,
-		Placement:         gofastly.NullString(vla.placement),
-		ResponseCondition: gofastly.NullString(vla.responseCondition),
+		ServiceID:         serviceID,
+		ServiceVersion:    serviceVersion,
+		Name:              df["name"].(string),
+		BucketName:        df["bucket_name"].(string),
+		User:              df["user"].(string),
+		AccessKey:         df["access_key"].(string),
+		PublicKey:         df["public_key"].(string),
+		GzipLevel:         uint(df["gzip_level"].(int)),
+		MessageType:       df["message_type"].(string),
+		Path:              df["path"].(string),
+		Region:            df["region"].(string),
+		Period:            uint(df["period"].(int)),
+		TimestampFormat:   df["timestamp_format"].(string),
+		Format:            vla.format,
+		FormatVersion:     uintOrDefault(vla.formatVersion),
```

Comment:
@phamann this function is called by multiple tests `TestAccFastlyServiceV1_logging_cloudfiles_basic` and `TestAccFastlyServiceV1_logging_cloudfiles_basic_compute` and to get the latter C@E test to pass I had to change the code to use `uintOrDefault` (as the fake terraform hcl code had omitted `format_version = 2` unlike the VCL equivalent which did set it, and also because there is a function `getVCLLoggingAttributes` which only sets `vla.formatVersion` if a VCL service type). 

I stumbled around on that failing test for a while and so I'd like you to give some extra attention to this change just so we can be sure it's correct. 

My concern is that if `vla.formatVersion` is not set when a C@E service, then it'll be `nil` and in the case of the pre-exising `uintOrDefault()` function, the 'default value' it returns is `0` but `FormatVersion` _should_ have a default value of `2` as its default (as per the api docs that state it refers to the apache log format version 2, where the actual acceptable values are 1 or 2, not zero!). 

I was writing my own code to check for nil and to set it to a default of `2` but then I noticed some other code also using `uintOrDefault()` when dealing with `FormatVersion` 🤔

Reply:
examples using `uintOrDefault()` with `FormatVersion`: https://github.com/fastly/terraform-provider-fastly/commit/7844ddb0161ff5518a5e0bc67a51b8261e903793 

Maybe there are no 'compute' tests that validate those particular logging blocks 🤔

Reply:
Patrick's response (for posterity):

> The `format_version` field is completely irrelevant for C@E services (as there is no VCL to format) so it doesn’t matter if the value is 0, 2 or nil for those services and hence why the C@E code paths don’t set or test for its existence.